Remove JMeter based performance tests (#1981)
hashgraph/hedera-mirror-node: hedera-mirror-test/README.md

# Acceptance Tests

This module covers the end-to-end (E2E) testing strategy employed by the mirror node for key scenarios.

## Overview

> In an effort to quickly confirm product capability during deployment windows, we desire to have E2E tests that would

Author: Fixed

> allow us to confirm functionality on core scenarios that spanned the main and mirror networks. HCS specifically is a key

Author: Actually already had that fixed. You must've reviewed a stale copy.

> allow us to confirm functionality on core scenarios that spanned the main and mirror networks. HCS specifically is a key

Reviewer: Yep, forgot to remove that comment after it got updated.

scenario where transactions are submitted to the main network, the mirror node importer ingests these to the database,
and the client subscribes to the mirror node gRCP API to receives messages in consensus order.

HTS is another key scenario where transactions are submitted to the main network, the mirror node importer ingests these
to the DB, and the mirror node REST API is periodically invoked to verify transactions. This E2E suite gives us the
ability to execute scenarios as regular clients would and gain the required confidence during deployment.

> To achieve this the tests utilize the [Hedera Java SDK](https://github.com/hashgraph/hedera-sdk-java) under the hood.

Reviewer 2: nit: missing Scheduled scenario as a key scenario.
Can either add it or we can simple start listing the services and scenario since the ingestion and checking with REST API's and network calls is the same approach for all

Author: To keep from having to maintain an exhaustive list of acceptance test scenarios I've made this more generic and do not mention HCS or HTS.

# Acceptance Tests

This module covers the end-to-end (E2E) testing strategy employed by the mirror node for key scenarios.

## Overview

In an effort to quickly confirm product capability during deployment windows, we desire to have E2E tests that would
allow us to confirm functionality for core scenarios that span the main and mirror networks. In general, the acceptance
tests will submit transactions to the mainnet scenario where transactions are submitted to the main network, the mirror
node importer ingests these to the database, and the client subscribes to either the mirror node gRPC or REST API to
receive results.

To achieve this the tests utilize the [Hedera Java SDK](https://github.com/hashgraph/hedera-sdk-java) under the hood.
This E2E suite gives us the ability to execute scenarios as regular clients would and gain the required confidence
during deployment.

## Cucumber

> A [BDD](https://cucumber.io/docs/bdd/) approach is desired for our E2E test strategy as it ensures we closely track
> valid customer scenarios. Cucumber is one framework that provides tools to follow this methodology. One benefit being
> that tests can be written in human-readable text. This allows developers, project managers and quality assurance to
> formulate tests that are connected to code to run valid customer scenarios. [Cucumber](https://cucumber.io) uses the
> Gherkin plain language parser to describe tests.

Reviewer: Just an idea, this section reads as a bit choppy, I realize it's been that way:
```suggestion
Our E2E tests use the [Cucumber](https://cucumber.io) framework, which allows them to be written in a [BDD](https://cucumber.io/docs/bdd/) approach that ensures we closely track valid customer scenarios.  The framework allows tests to be written in a human-readable text format by way of the Gherkin plain language parser, which gives developers, project managers, and quality assurance the ability to write easy-to-read scenarios that connect to more complex code underneath.
```

Author: Copied that

```

#### Feature Tags

> Tags: Tags allow you to filter which cucumber scenarios and files are run. By default tests marked with the @sanity tag

Reviewer: nit:
```suggestion
Tags: Tags allow you to filter which Cucumber scenarios and files are run. By default tests marked with the @sanity tag
```

Author: Uppered it

```

#### Feature Tags

Tags: Tags allow you to filter which cucumber scenarios and files are run. By default tests marked with the @Sanity tag
are run. To run a different set of files different tags can be specified

Acceptance test cases

> `./mvnw clean integration-test --projects hedera-mirror-test/ -P=acceptance-tests -Dcucumber.filter.tags="@Acceptance"`

Reviewer 2: we'll remove the upper case tags soon
```suggestion
`./mvnw clean integration-test --projects hedera-mirror-test/ -P=acceptance-tests -Dcucumber.filter.tags="@acceptance"`
```

Author: Changed all tags in README to lowercase

All cases

> `./mvnw clean integration-test --projects hedera-mirror-test/ -P=acceptance-tests -Dcucumber.filter.tags="@FullSuite"`

Reviewer 2: ```suggestion
`./mvnw clean integration-test --projects hedera-mirror-test/ -P=acceptance-tests -Dcucumber.filter.tags="@fullsuite"`
```

Negative cases

> `./mvnw clean integration-test --projects hedera-mirror-test/ -P=acceptance-tests -Dcucumber.filter.tags="@Negative"`

Reviewer 2: ```suggestion
`./mvnw clean integration-test --projects hedera-mirror-test/ -P=acceptance-tests -Dcucumber.filter.tags="@negative"`
```

Edge cases

> `./mvnw clean integration-test --projects hedera-mirror-test/ -P=acceptance-tests -Dcucumber.filter.tags="@Edge"`

Reviewer 2: ```suggestion
`./mvnw clean integration-test --projects hedera-mirror-test/ -P=acceptance-tests -Dcucumber.filter.tags="@edge"`
```
